internal/coreinternal/traceutil/traceutil.go

```diff
@@ -42,6 +42,22 @@ func SpanKindStr(sk ptrace.SpanKind) string {
 	return ""
 }
 
+func SpanKindFromStr(sk string) ptrace.SpanKind {
+	switch sk {
+	case "SPAN_KIND_INTERNAL":
+		return ptrace.SpanKindInternal
+	case "SPAN_KIND_SERVER":
+		return ptrace.SpanKindServer
+	case "SPAN_KIND_CLIENT":
+		return ptrace.SpanKindClient
+	case "SPAN_KIND_PRODUCER":
+		return ptrace.SpanKindProducer
+	case "SPAN_KIND_CONSUMER":
+		return ptrace.SpanKindConsumer
```

Comment:
Why support these representations? We use shorter forms https://github.com/open-telemetry/opentelemetry-collector/blob/main/pdata/ptrace/span_kind.go#L50-L64 in the collector

Reply:
I went with these to be backwards compatible with filterspan. At the moment anyone filtering by span kind is using these representation and not pdata's. 

The bridge can probably do it, so if we're worried about a deprecation process I can look into mapping the string values coming in from the filter configs to the pdata representation instead.

Reply:
Maybe we can support both for now. Then we can deprecate the old representation, once the old filter processor config interface is deprecated. Is there a way to print deprecation warnings when reading OTTL statements?

Reply:
> The bridge can probably do it, so if we're worried about a deprecation process I can look into mapping the string values coming in from the filter configs to the pdata representation instead.

After giving it a second thought, I think this is actually a cleaner path forward

Reply:
@dmitryax switched.  I will make it work in the bridge.

Reply:
@dmitryax regex support in the existing `filterspan` interface makes it so a simple map in the bridge between `SPAN_KIND_INTERNAL` and `Internal` won't be enough.  Users are allowed to do stuff like

```yaml
filter/spans:
  spans:
    include:
      match_type: regex
      span_kinds:
        - SPAN_KIND_
```

Which, while very silly, would still break if OTTL only supports the pdata `String()` values.  In order to be backwards compatible I believe we need to add support for the `SPAN_KIND_*` strings, but to make the transition easier for OTTL users in the future I'll add it as a different accessor.